src/index.ts

```diff
@@ -70,67 +76,82 @@ export class Api {
             Authorization: `Basic ${this.basicAuthToken}`,
           },
         },
-      )
-      .then(({ data }) => {
-        this.setTokenHeaders(data);
-        this.refreshSessionJustBeforeExpiry();
-      });
-
-    this.updatePrerequest(request);
+      );
+      this.setTokenHeaders(data);
+    } catch (error) {
+      console.error('Requesting new OHIP session token failed', error);
+      this.clearTokens();
+      throw error;
+    }
   }
 
-  private refreshSessionJustBeforeExpiry() {
-    if (!this.tokenExpiration) return;
-
-    this.refreshTimeout = setTimeout(() => {
-      if (!this.refreshToken) {
-        this.initializeSession();
-        return;
-      }
-
-      const request = this.clientDict.oauth.tokens
-        .getToken(
-          {
-            // @ts-ignore
-            grant_type: 'refresh_token',
-            refresh_token: this.refreshToken,
-          },
-          {
-            type: ContentType.UrlEncoded,
-            headers: {
-              Authorization: `Basic ${this.basicAuthToken}`,
-            },
+  private async renewAuthToken() {
+    try {
+      const { data } = await this.clientDict.oauth.tokens.getToken(
+        {
+          // @ts-ignore
+          grant_type: 'refresh_token',
+          refresh_token: this.refreshToken,
+        },
+        {
+          type: ContentType.UrlEncoded,
+          headers: {
+            Authorization: `Basic ${this.basicAuthToken}`,
           },
-        )
-        .then(({ data }) => {
-          this.setTokenHeaders(data);
-          this.refreshSessionJustBeforeExpiry();
-        });
-
-      this.updatePrerequest(request);
-    }, this.tokenExpiration * 900);
+        },
+      );
+      this.setTokenHeaders(data);
+    } catch (error) {
+      console.error('Refreshing OHIP session token failed', error);
+      await this.requestNewAuthToken();
+    }
   }
 
   private setTokenHeaders(response: OAuth2TokenResponse) {
     // @ts-ignore
     this.refreshToken = response.refresh_token;
```

Comment:
Is this ignored because the response contains refresh_token already but not yet updated on the generated types for `OAuth2TokenResponse `?

Reply:
yup